### PR TITLE
fix(err): fix distinct_id, set personless and use a uuid

### PR DESCRIPTION
**Pull request by oliverb123**
Ingest will punish you for using a fixed value for distinct_id, so let's not do that.

### CODE REVIEW

File: posthog/client.py [hunk 1]
```diff
@@ -362,7 +362,7 @@ def page(
     def capture_exception(
         self,
         exception=None,
-        distinct_id=DEFAULT_DISTINCT_ID,
```
Comment (daibhin) [inline posthog/client.py:365]: We should remove this variable entirely

File: posthog/exception_capture.py [hunk 2]
```diff
@@ -61,14 +62,15 @@ def exception_receiver(self, exc_info, extra_properties):
...
 
-            distinct_id = metadata.get("distinct_id") if metadata else DEFAULT_DISTINCT_ID
-            # Make sure we have a distinct_id if its empty in metadata
-            distinct_id = distinct_id or DEFAULT_DISTINCT_ID
+            # if there's no distinct_id, we'll generate one and set personless mode
+            # via $process_person_profile = false
+            properties = {}
+            if distinct_id is None:
+                properties["$process_person_profile"] = False
+                distinct_id = uuid.uuid4()
```
Comment (daibhin) [inline posthog/exception_capture.py:72]: Should we be using v7 going forward? Unrelated to this change

**Review by daibhin — APPROVED** [review]

File: posthog/exception_capture.py (hunk 2 above)
Comment (oliverb123) [reply posthog/exception_capture.py:72]: Maybe? We don't really care what's in `distinct_id`, and don't make recommendations to customers about it I think - we probably should us v7 everywhere just for the sake of the habit, but the python standard library doesn't support it yet afaict.

**Comment by oliverb123** [comment]
I ran `black .` locally and then committed exactly what it gave me, and CI says it's wrong. Should I care about lint checks?

File: posthog/client.py (hunk 1 above)
Comment (oliverb123) [reply posthog/client.py:365]: Done

**Comment by daibhin** [comment]
@oliverb123 I think we use ruff instead. My VSCode is just setup to auto format afaik

**Pull request merged**